### ISSUE TITLE
Magento ignores defined email template type in outgoing emails

### ISSUE THREAD

**Issue by jakwinkler** [magento/magento2#30808]
### Preconditions (*)

1. Magento 2.3.5
2. Create a new email template as text/plain 

### Steps to reproduce (*)

1. Create new email template `type="text"`
2. Create a new module with CLI method  to send that email template
3. Send out template 

### Expected result (*)

1. Email Content Type should be `text/plain`

### Actual result (*)

1. Email content type is ALWAYS `text/html`

![image](https://user-images.githubusercontent.com/4256191/98271507-4cd96780-1f90-11eb-8e49-c8d9fb95ecf5.png)

### Source of the issue:
`\Magento\Framework\Mail\Template\TransportBuilder`  class has a **bug** in method `prepareMessage`.
Email type variable is not passed anywhere in the code, it is just listed in switch statement

### Solution:
Change this:
```
$this->messageData['body'] = $this->mimeMessageInterfaceFactory->create(
            ['parts' => [$mimePart]]
        );
```

into this:

```
/** @var \Magento\Framework\Mail\MimePartInterface $mimePart */
        $mimePart = $this->mimePartInterfaceFactory->create(
            [
                'content' => $content,
                'type' => $part['type']
            ]
        );
```

buggento for the win!

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by qsolutions-pl** [comment]
This was fixed in 2.3.6 but fix was not ported to ealier versions.

**Comment by qsolutions-pl** [comment]
Closing the issue there is no point in discussion anymore.
